salt/modules/boto_iam.py

```diff
@@ -740,8 +740,34 @@ def delete_group(group_name, region=None, key=None,
         return False
 
 
-def create_login_profile(user_name, password, region=None, key=None,
-                         keyid=None, profile=None):
+def get_login_profile(user_name, region=None, key=None, keyid=None, profile=None):
+    '''
+    Return current Login Profile attached to user, if the user exists and
+    has a profile attached, None otherwise.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_iam.get_login_profile bob.dobbs
+    '''
+    conn3 = __utils__['boto3.get_connection']('iam', region=region, key=key,
+                      keyid=keyid, profile=profile)
+    try:
+        ret = conn3.get_login_profile(UserName=user_name)
+        ret.pop('ResponseMetadata')
+        return ret
+    except botocore.exceptions.ClientError as e:
+        log.debug(e)
+        err = e.response.get('Error', {}).get('Message')
+        # Error.Code is 'NoSuchEntity' for both a missing user and a missing Login Profile...
+        if 'Cannot find Login Profile for User' not in err:
+            log.error('Failed to get Login Profile for IAM user {0}: {1}'.format(user_name, err))
+        return None
+
+
+def create_login_profile(user_name, password, password_reset_required=False,
```

Comment:
Addubg `password_reset_required` in the beginning of the kwargs list will cause an  API change. Can you move that to the end of the list please?

Reply:
I actually tracked down every usage of create_login_profile() extant in the code-base (there was only one - in the boto_iam state module) and updated to match this param ordering.

I can see your point vis-a-vis hand-jammed module function calls though.  I hate the inelegance of shoving a new password related param on the end, far away from `password` itself but I see where you're coming from.